File: Data/Game.cs
using Microsoft.EntityFrameworkCore;

namespace rock_paper_scissors.Data
{
    public class Game
    {
        public Guid Id { get; set; }
        public GameSession Player1 { get; set; }
        public GameSession? Player2 { get; set; }

    }
    [Owned]
    public class GameSession
    {
        public string Nick { get; set; }
        public Weapon? WeaponRound1 { get; set; }
        public Weapon? WeaponRound2  { get; set; }
        public Weapon? WeaponRound3  { get; set; }
        public Weapon? WeaponRound4  { get; set; }
        public Weapon? WeaponRound5  { get; set; }
    }

}

File: ClientApp/src/app/models/game.ts
export type GameEvent = 'WaitingForSecondPlayerToJoin'
        | 'StartTheGame'
        | 'WaitingForPlayerToPlay'
        | 'RoundEnd'
        | 'GameEnd'
        | 'AskServerResponse';

export type GameStatus = 'NotStarted'
        | 'Initiated'
        | 'InProgress'
        | 'Finished';

export type Weapon = 'Rock'
        | 'Paper'
        | 'Scissors';

export type GameResult = 'Win'
        | 'Lose'
        | 'Draw';

export interface Game {
  status: GameStatus;
  roundResult: GameResult | null;
  gameResult: GameResult | null;
  round: number;
}

export interface GameSession {
  sessionId: string;
  nick: string;
  weaponRound1: Weapon;
  weaponRound2: Weapon;
  weaponRound3: Weapon;
  weaponRound4: Weapon;
  weaponRound5: Weapon;
}

export interface FinalRoundResult {
  playerWeapon: Weapon;
  opponentWeapon: Weapon;
  playerResult: GameResult;
 }

export interface FinalGameStatistic {
  playerNick: string;
  opponentNick: string;
  playerFinalResult: GameResult;
  round1Result: FinalRoundResult;
  round2Result: FinalRoundResult;
  round3Result: FinalRoundResult;
  round4Result: FinalRoundResult;
  round5Result: FinalRoundResult;
}
